site/app/controllers/PollController.php

```diff
@@ -152,7 +154,14 @@ public function addNewPoll() {
                 $answers[] = $_POST["option_id_" . $i];
             }
         }
-        $this->core->getQueries()->addNewPoll($_POST["name"], $_POST["question"], $responses, $answers, $_POST["release_date"], $orders);
+        $poll_id = $this->core->getQueries()->addNewPoll($_POST["name"], $_POST["question"], $responses, $answers, $_POST["release_date"], $orders);
+        $file_path = null;
+        if (array_key_exists("image_file", $_FILES) && $_FILES["image_file"]["name"] != "") {
+            $file = $_FILES["image_file"];
+            $file_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "uploads", "polls", "poll_image_" . $poll_id . "_" . $_FILES["image_file"]["name"]);
```

Comment:
Why is maintaining the name of the file important on the filesystem?

Reply:
This was my suggestion.  Eventually when we export the images associated with polls, it will be helpful to have the original names as part of the filenames, they might be meaningful to the instructor.

Reply:
That's fair. I would greatly prefer that we utilize the new `image_path` DB field to store the name that an instructor uses and that on the filesystem itself we just store it as `FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "uploads", "polls", "poll_image_" . $poll_id)`. On export, we can provide the original name from the DB, while on the filesystem itself, we do not have to worry about the instructor uses god knows what characters.

Reply:
Sounds good to me

Reply:
@MasterOdin I talked to Barb about this and she suggested we keep the current format but remove any special characters from the provided name first (just allow letters, numbers, some characters like _, -, etc). Do you think this makes sense?

Reply:
So the philosophy I would cater towards is that it's always better to store a filename within a database and then use a simple, unique identifier for the file on-disk (or in this case `"poll_image_" . $poll_id` which we let get overwritten on each new upload). It simplifies certain elements of handling files (e.g. on upload, we can remove the explicit `unlink` step for the old file, and just have the filesystem overwrite on copy) and in a more generic world where an upload folder is more flat, upload name collision attacks.

This isn't a hill I'm going to die on though, and if you or @bmcutler feel like you want to have the name on disk, so be it.